streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -791,12 +788,16 @@ public boolean conditionMet() {
 
         final ReadOnlyKeyValueStore<String, String> store2 = kafkaStreams.store(storeName, QueryableStoreTypes.<String, String>keyValueStore());
 
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return "125".equals(store2.get("a")) && "34".equals(store2.get("b"));
-            }
-        }, maxWaitMs, "wait for agg to be '123'");
+        try {
+            TestUtils.waitForCondition(new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    return store2.get("a").endsWith("125") && store2.get("b").endsWith("34");
+                }
+            }, maxWaitMs, "wait for agg to be <a,125> and <b,34>");
```

Comment:
Instead of loosen the check here, could we wait until one commit has been executed before `throw new RuntimeException("Injected test exception");`?